### PR TITLE
[FCE-798] Bump to Expo SDK 52

### PR DESCRIPTION
**Pull request by MiloszFilimowski**
## Description

- Bumped example deps to support Expo SDK 52
- Updated peer deps to require Expo 52
- Migrated EventEmitter to new API

## Motivation and Context

- We can now support Expo SDK 52

## How has this been tested?

- Run and tested on Android/iOS

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)

### CODE REVIEW

File: packages/react-native-client/src/RNFishjamClientModule.ts [hunk 1]
```diff
@@ -80,7 +80,25 @@ type RNFishjamClient = {
...
+  Reconnected: 'Reconnected',
+  Warning: 'Warning',
+  PeerStatusChanged: 'PeerStatusChanged',
+  CurrentCameraChanged: 'CurrentCameraChanged',
+} as const;
 
-export default nativeModule as RNFishjamClient & NativeModule;
+export default requireNativeModule('RNFishjamClient') as RNFishjamClient &
+  NativeModule<
+    // TODO: Make event arguments typesafe instead of generic.
```
Comment (MiloszFilimowski) [inline packages/react-native-client/src/RNFishjamClientModule.ts:102]: **note:** Since we now can typecheck all events I'd do it in a separate PR so it's easier to review.

**Review by maksg — APPROVED** [review]

**Review by maksg — APPROVED** [review]

**Comment by MiloszFilimowski** [comment]
Tested with an Expo 51 app and works fine.
